Source/WebKit/PlatformGTK.cmake

```diff
@@ -389,6 +390,7 @@ GENERATE_GLIB_API_HEADERS(WebKit WebKitGTK_HEADER_TEMPLATES
     "-DWTF_PLATFORM_GTK=1"
     "-DWTF_PLATFORM_WPE=0"
     "-DUSE_GTK4=$<BOOL:${USE_GTK4}>"
+    "-DENABLE_WK_WEB_EXTENSIONS=$<BOOL:${ENABLE_WK_WEB_EXTENSIONS}>"
```

Comment:
Nooo don't do this! Having conditionally-compiled public APIs will be confusing and not fun. Let's keep things fun. :)

What we should do instead is always build all of the APIs, and leave them as stubs in case `ENABLE_WK_WEB_EXTENSIONS` is not defined. Generally you would return a default value (e.g. 0 or empty string) from functions that return values, and otherwise do nothing. You can use `ENABLE_WK_WEB_EXTENSIONS` in the source code, but not in the public headers. Since you won't use it in the headers, you don't need to pass it to unifdef here.

(We should also consider -- separately -- whether `ENABLE_WK_WEB_EXTENSIONS` is really needed at all. Unless it's necessary to guard use of some external dependency on Apple platforms, then it can probably be removed in favor of using a runtime feature flag.)

Reply:
We need the define for Apple platforms.

Reply:
If the define is needed, then I can add stub functions. I don't love having a useless API in builds, especially if it's not immediately clear that it's useless, but if it's done elsewhere that's fine.

Reply:
You could also just enable the setting (`WEBKIT_OPTION_DEFAULT_PORT_VALUE` in OptionsGTK.cmake and possibly OptionsWPE.cmake) if ready to do so.

Reply:
I'd rather not enable the setting just yet, since there's still a bit more work to make it useful. I do hope to ship this soon, or at least ship some of the APIs, so I'm hoping we don't have to revert this for a stable branch.

Reply:
it would be nice if I could remmeber why I made a change.

these lines in PlatformGTK and PlatformWPE weren't for enabling the API, it was for properly creating the error enumeration types (or not, if the API was disabled). The points about conditional API creation are valid, but without these lines, the error enumerations have to always exist as glib-mkenums doesn't recognise that they're conditionally enabled.

Reply:
I'm also not a fan of having the symbols (and header) enabled or not depending on a build time setting. We should decide if we want the feature, and act accordingly. We are landing this on `main`, which is where in-development features happen, so IMO it's fine to have it unconditionally enabled.

Later on, when doing the next stable branch, we can decide to remove the public API (also unconditionally) _only in the branch_ if the feature is not considered ready for public consumption.

Reply:
I had fixed this on a local branch, where the enumerations this line enabled are just always enabled. Apparently I forgot to push it though